.build/docker/docker-entrypoint.sh

```diff
@@ -0,0 +1,53 @@
+#!/bin/bash
+
+# Initialize flags for config.json and private key
+CONFIG_EXISTS=false
+PK_EXISTS=false
+
+# Check if config.json exists
+if [ -f /app/config.json ]; then
+  echo "config.json detected"
+  CONFIG_EXISTS=true
+else
+  echo "config.json not found"
+fi
+
+# Check if data-proxy-private-key.json exists
+if [ -f /app/data-proxy-private-key.json ]; then
+  echo "data-proxy-private-key.json detected"
+  PK_EXISTS=true
+elif [ -n "$SEDA_DATA_PROXY_PRIVATE_KEY" ]; then
+  # If private key file does not exist, check if the private key is provided via environment variable
+  echo "Private key provided via environment variable"
+  echo "$SEDA_DATA_PROXY_PRIVATE_KEY" >/app/data-proxy-private-key.json
+  PK_EXISTS=true
+else
+  echo "No private key provided"
+fi
+
+run_bun_command() {
+  if ! bun "$@"; then
+    echo "Failed to run: bun $*"
+    exit 1
+  fi
+}
+
+# Determine the command to run based on the presence of config.json and private key
+if [ "$CONFIG_EXISTS" = true ] && [ "$PK_EXISTS" = true ]; then
+  # Both config.json and private key are provided
+  echo "Running with config and private key"
+  RUN_CMD="start run --config /app/config.json --private-key-file /app/data-proxy-private-key.json"
+elif [ "$CONFIG_EXISTS" = true ] && [ "$PK_EXISTS" = false ]; then
+  # Only config.json is provided
+  echo "Running with config only"
+  run_bun_command start init
+  RUN_CMD="start run --config /app/config.json"
```

Comment:
I think the binary takes care of these checks, is there a reason to also do it in the entrypoint?

Reply:
I'm not entirely sure what the best approach would be here... 🤔 

I'm partial to letting the binary take care of everything as this reduces the maintenance burden on our end since the container doesn't really need to know anything.

If I want to initialise I should do `docker run <IMAGE_NAME> init` and pass a volume so I can persist the config and generated key.
For registration I can then do `docker run <IMAGE_NAME> register <PAYOUT_ADDRESS> <AMOUNT>` and click the link from the output to complete the registration.
And to run it just do `docker run <IMAGE_NAME> run`.

I guess that get a little messy with compose though, don't know how to do that in an elegant fashion.